src/wpool_process_sup.erl

```diff
@@ -34,6 +34,7 @@ start_link(Parent, Name, Options) ->
 init({Name, Options}) ->
   Workers = proplists:get_value(workers, Options, 100),
   Strategy = proplists:get_value(strategy, Options, {one_for_one, 5, 60}),
+  _ = maybe_add_event_handler(Options),
```

Comment:
Since we're using `lists:foreach/2` now, you don't need to discard the result anymore ;)
```suggestion
  maybe_add_event_handler(Options),
```